src/files/ocsigenserver.conf/gen.ml

```
@@ -0,0 +1,158 @@
+
+    <logdir>_LOGDIR_</logdir>
+    <datadir>_DATADIR_</datadir>
+    _COMMANDPIPE_
+    _MIMEFILE_
+
+    <charset>utf-8</charset>
+
+    <findlib path="_METADIR_"/>_FINDLIBEXTRA_
+
+    <extension findlib-package="_EXTPACKAGENAME_.staticmod"/>
+
+    <!-- Inclusion of all external configuration files matching *.conf
+         from the directory 'dir' (in alphabetical order): -->
+    <extconf dir="_CONFIGDIR_/conf.d" />
+
+    <host charset="utf-8" hostfilter="*">
+
+      <site path="ocsigenstuff" charset="utf-8">
+        <static dir="_STATICPAGESDIR_/ocsigenstuff" />
+      </site>
+
+      <static dir="_STATICPAGESDIR_" />
+
+    </host>
+
+  </server>
+
+</ocsigen>|}
+
+let conf_ml =
+  {|(* Warning: this file has been generated - DO NOT MODIFY! *)
+
+let version_number = _VERSION_
+let config_file = ref _CONFIGFILE_
+let is_native = Sys.backend_type = Sys.Native
+let logdir = ref (Some _LOGDIR_)
+let mimefile = ref _MIMEFILE_
+let datadir = ref _DATADIR_
+let bindir = ref _BINDIR_
+let extdir = ref _EXTDIR_
+let command_pipe = ref _COMMANDPIPE_
+let builtin_packages =
+  List.fold_left
+    (fun a s -> Ocsigen_lib.String.Set.add s a)
+    Ocsigen_lib.String.Set.empty
+    [_DEPS_]|}
+
+let interpolate f s =
+  let regexp = Str.regexp "_\\([A-Z]+\\)_" in
+  let f s = f (Str.matched_group 1 s) in
+  Str.global_substitute regexp f s
+
+open Options
+
+let libdir () =
+  if libdir_set = 0
+  then (
+    let inp = Unix.open_process_in "ocamlfind printconf destdir" in
+    let libdir = input_line inp in
+    ignore (Unix.close_process_in inp);
+    libdir)
+  else libdir
+
+let deps () =
+  let extra_deps =
+    [ "ocsigenserver.polytables"
+    ; "ocsigenserver.cookies"
+    ; "ocsigenserver.baselib.base"
+    ; "ocsigenserver.baselib"
+    ; "ocsigenserver.http"
+    ; "ocsigenserver" ]
+  in
+  let packages =
+    "lwt_ssl,bytes,lwt.unix,logs,logs-syslog,syslog-message,ipaddr,findlib,cryptokit,re,str,xml-light,dynlink,cohttp-lwt-unix,hmap"
```

This needs to be updated as follows:
```suggestion
    "lwt_ssl,bytes,lwt.unix,logs,logs-syslog.unix,syslog-message,ipaddr,findlib,cryptokit,re,str,xml-light,dynlink,cohttp-lwt-unix"
```
I suspect the computation of the dependencies fail silently when `hmap` is not installed. This is why `ocsigenserver` tries (unsuccessfully) to load `unix.cma`. One should check the return value of `Unix.close_process_in inp` below.

---

I added a failure case when this happens so it should now be noticed. Thanks for the debugging !